packages/nextjs/pages/blockexplorer/address/[address].tsx

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useMemo, useState } from "react";
+import { useRouter } from "next/router";
+import { ethers } from "ethers";
+import fs from "fs";
+import { GetServerSideProps } from "next";
+import path from "path";
+import { ParsedUrlQuery } from "querystring";
+import * as chains from "wagmi/chains";
+import {
+  AddressCodeTab,
+  AddressLogsTab,
+  AddressStorageTab,
+  PaginationButton,
+  TransactionsTable,
+} from "~~/components/blockexplorer/";
+import { Address, Balance } from "~~/components/scaffold-eth";
+import deployedContracts from "~~/generated/deployedContracts";
+import { useFetchBlocks } from "~~/hooks/scaffold-eth";
+
+type AddressCodeTabProps = {
+  bytecode: string;
+  assembly: string;
+};
+
+type PageProps = {
+  address: string;
+  contractData: AddressCodeTabProps | null;
+};
+
+const AddressPage = ({ address, contractData }: PageProps) => {
+  const router = useRouter();
+  const { blocks, transactionReceipts, currentPage, totalBlocks, setCurrentPage, isLoading } = useFetchBlocks();
+  const [activeTab, setActiveTab] = useState("transactions");
+  const [isContract, setIsContract] = useState(false);
+
+  const provider = useMemo(() => new ethers.providers.JsonRpcProvider("http://localhost:8545"), []);
```

Comment:
We shouldn't hardcode the JSON provider here.

We have the `getLocalProvider` util

Reply:
(same for other places where you do the same)

Reply:
I tried to use getLocalProvider but still added JsonRpcProvider as fallback because getLocalProvider may return undefined and that causes bunch of other issues.
Let me know if there are any better solutions because I couldn't find one.